src/playcanvas-gltf.js

```diff
@@ -141,19 +141,44 @@ Object.assign(window, function () {
                 }
             } else {
                 // translation, rotation or scale
-                keyType = AnimationKeyableType.NUM;
+
```

Comment:
For path is not 'weights', one channel only has one corresponding AnimationCurve. Key on that AnimationCurve could be a single number (NUM), or a 3d vector (VEC) or a 4d quaternion (QUAT). But there is only one curve, animating 1d, 3d or 4d data together. 

So here the logic seems is depending on the DOF (degree of freedom) of animation data, decide whether the curve should have either NUM, VEC or QUAT key.  The name numCurves is a bit confusing. The logic is good.

Reply:
renamed to `valueSize` in  https://github.com/SG-Oleksandr/playcanvas-gltf/commit/ec6063e6f4227b99f58cc927251ad45cc5cadb4f